cli/tests/end_to_end/extension.rs

```diff
@@ -75,6 +75,33 @@ pub async fn get_groups() {
         .stdout(predicates::str::contains("groups"));
 }
 
+#[tokio::test]
+pub async fn create_and_delete_project() {
+    let test_cli = TestCli::builder().with_config(None).build();
+
+    test_cli
+        .extension(
+            r#"
+            try {
+                await PhylumApi.deleteProject("create_and_delete_test_project")
+            } catch (e) {
+            }
+
+            let projectId = await PhylumApi.createProject("create_and_delete")
+            let duplicatedProjectId = await PhylumApi.createProject("create_and_delete")
```

Comment:
I'm not sure we want to just silently ignore duplicates? In a follow-up I'd want to introduce error handling, but that doesn't mean it shouldn't fail at all right now? I wouldn't want to change an API from non-failing to failing, that seems kinda inconsistent?

An endpoint to get project info might help with that, just to see if it needs to be created or not ahead of time. I don't think we have one of those yet but it probably would make sense? We have `getCurrentProject` already.

Reply:
I agree that a breaking change in an API from non-failing to failing would be bad. But, as per the previous discussion, I think the way forward would make the "created" and "already exists" both _successful_ states; if anything, those two states would optionally be different in the response format (e.g. carry an "existing" flag or something similar alongside the id), but I don't even think the user would benefit from that.

I think erroring out on a project existing is bad from the user's perspective, as it forces them to do error management when all they care about is either creating or retrieving a project. The use case for checking whether a project exists would be covered by a "get project by name" API -- which we already have in the CLI, just not exposed to extensions. Once we have that, the user has no benefit in manually checking for the existence of a project, unless they need to for purposes different than deciding whether to create a project or not.

Also, "duplicated" here is a misnomer: a project that was already existing is returned with its id, rather than copied into a new project.

Given the above, I think the current solution is appropriate, but I'm open to other feedback.

An argument in favor of simplifying this API would be to more faithfully mimick our server side API, but I don't think that is necessarily an added value over providing a more ergonomic API to the users.

Reply:
That makes sense; the only point I question there is whether the user ever cares about actually knowing the project's state. I think not: all of the operations that can be done on a project (submitting an analysis to it; querying its information; deleting it) do not need to know anything about its state beforehand, and will error out if the project doesn't exist, as intended. Whatever we decide to name it, I think the semantics of "create project or return project handle" is the one that is going to be more in line with what a user instinctively does: 
- if someone cares about only creating a project if it doesn't exist, they can check manually with other APIs
- if someone _does not_ care about the state of the project, they don't have extra state to manage, and an error will have to be handled as an error
- if someone is not aware of this mechanism, they won't notice, and nothing bad is going to happen; this is going to be true also in the augmented response case where we might have something like `{ "id": "abcd1234", "status": "created" }` vs `{ "id": "abcd1234", "status": "existing" }`

On the other hand, returning the project to a clean slate could actually be what surprises a user, who might have their analysis canceled.

Considering this, what would you suggest a good way forward is? Pinging also @kylewillmon for feedback as he raised the issue originally.

Reply:
First, we should take the return value as a separate discussion. The Project ID is basically unused in our extension API because `analyze` take a Project Name rather than an ID. This function may not need a return value at all.

> it's basically just an `ensureProjectExists` method at that point rather than a `createProject` endpoint.

Renaming this function might solve all our problems. Ensuring that a project exists is the [primary use case](https://github.com/phylum-dev/community-extensions/blob/c26f325afa88607e2f9f3df7af7ba18d19188655/build-gradle/main.ts#L245-L272) for this feature.

Creating a clean API for `createProject` would probably require that we resolve #719 first. That problem does not exist with `ensureProjectExists`.

Reply:
It's just that `ensureProjectExists` is a little less RESTy/CRUDy, at least it feels a little bit weird to me.

I don't want to block this PR on just this question, but I'd definitely say there's only two good choices:
 - We never want this to return an error, even after #719, then we can just keep it as-is
 - We want it to return a custom error after #719, then I think we should already return an error

I'd at least like a clear decision for one of the two, my understanding right now is that it's basically a mix of the two which I don't really like.

Reply:
Yeah... We might need to go back to the issue (#698) and discuss the details of the API before continuing to go back and forth on this PR...